<launch>

  <arg name="map_frame_id" default="map"/>
  <arg name="map_size_x" default="20"/>
  <arg name="map_size_y" default="20"/>
  <arg name="map_size_z" default="10"/>

  <node pkg ="param_env" name ="structure_map" type ="structure_map" output = "screen">    
    <!-- this is the map init position, not the odom-->

    <param name="map/x_size"     value="$(arg map_size_x)" />
    <param name="map/y_size"     value="$(arg map_size_y)" />
    <param name="map/z_size"     value="$(arg map_size_z)" />
    <param name="map/x_origin"   value="0.0" type="double"/>
    <param name="map/y_origin"   value="0.0" type="double"/>
    <param name="map/z_origin"   value="0.0" type="double"/>


    <param name="map/resolution" value="0.2"/>
    <param name="map/frame_id"     value="$(arg map_frame_id)" />

    <!-- total obstacle ratios -->
    <param name="map/cylinder_ratio" value="0.05" type="double"/>
    <param name="map/circle_ratio"   value="0.01" type="double"/>
    <param name="map/gate_ratio"     value="0.01" type="double"/>
    <param name="map/ellip_ratio"    value="0.01" type="double"/>
    <param name="map/poly_ratio"     value="0.01" type="double"/>

    <param name="params/w1" value="0.05"/>
    <param name="params/w2" value="0.4"/>
    <param name="params/w3" value="0.6"/>
    <param name="params/w4" value="1.5"/>



  </node>


<!--   <arg name="rviz_config" default="$(find param_env)/launch/default.rviz"/>-->
<!--   <node pkg="rviz"-->
<!--     type="rviz"-->
<!--     name="rviz"-->
<!--     args="-d $(arg rviz_config)"/>-->


</launch>
